packages/@pollyjs/adapter-puppeteer/src/index.js

```diff
@@ -203,22 +191,25 @@ export default class PuppeteerAdapter extends Adapter {
         body: await response.text()
       };
     } catch (error) {
-      console.error(error);
       throw error;
     } finally {
       this[PASSTHROUGH_PROMISES].delete(requestId);
     }
   }
 
-  respondToRequest(pollyRequest) {
+  async respondToRequest(pollyRequest, error) {
     const { request } = pollyRequest.requestArguments;
     const { response } = pollyRequest;
 
-    request.respond({
-      status: response.statusCode,
-      headers: response.headers,
-      body: response.body
-    });
+    if (error) {
+      await request.abort();
```

Comment:
Might be worth leaving a comment on why we abort here.  It's not clear and realize it's code that was just moved in this PR.